maven-universe/src/main/java/org/jboss/galleon/universe/maven/MavenChannel.java

```diff
@@ -115,7 +115,22 @@ public Path resolve(FeaturePackLocation fpl) throws MavenUniverseException {
             artifact.setVersionRange(versionRange);
             producer.getRepo().resolveLatestVersion(artifact, getFrequency(fpl), versionIncludePattern, versionExcludePattern);
         } else {
-            artifact.setVersion(fpl.getBuild());
+            String build = fpl.getBuild();
```

Comment:
Couldn't we simply check whether the specified build matches the version patterns of this channel instead?

Reply:
@aloubyansky We could but that doesn't deal with the user providing a build that is outside the version range or frequency of the channel.